src/test/user.spec.js

```diff
@@ -37,7 +42,35 @@ describe('Test user login, signup and account verification', () => {
         });
     });
 
-    it('it should throw an error when user with an unverified account wants to signup', done => {
+    it('should mock a sendgrid api request', async () => {
+      nock('https://api.sendgrid.com')
+        .post('/api/mail.send.json')
+        .reply(200, {
+          message: 'success'
+        });
+
+      const res = await chai
+        .request('https://api.sendgrid.com')
+        .post('/api/mail.send.json');
+      expect(res.body.message).to.include('success');
+      expect(res.status).to.eql(200);
+    });
+
+    it('should mock a failed sendgrid api request', async () => {
```

Comment:
😅 
The test block should say what is done by the `API` being tested. Your `API` doesn't **mock** sendgrid. You api sends requests to sendgrid and **handles** both success and failure cases. That's what your test should say.

The assertions should demonstrate this as well.

Reply:
You may look at this for [ideas](https://github.com/segunolalive/helloBooks/blob/development/server/test/controllers/userController.spec.js#L191)